src/emulator/apphosting/serve.ts

```diff
@@ -167,3 +201,44 @@ export function getEmulatorEnvs(): Record<string, string> {
 
   return envs;
 }
+
+async function tripFirebasePostinstall(
+  rootDirectory: string,
+  env: Record<string, string>,
+): Promise<void> {
+  let firebaseUtilPath;
+  try {
+    firebaseUtilPath = dirname(
+      require.resolve("@firebase/util/package.json", { paths: [rootDirectory] }),
+    );
+  } catch (err: unknown) {
+    return;
+  }
+
+  await spawnWithCommandString("npm run postinstall", firebaseUtilPath, env).catch((err) => {
+    logWarning(`Failed to run postinstall script for @firebase/util: ${getError(err).message}.`);
+  });
+}
+
+async function getBackendAppConfig(
+  projectId: string,
+  backendId: string,
+): Promise<WebConfig | undefined> {
+  if (Constants.isDemoProject(projectId)) {
+    return constructDefaultWebSetup(projectId);
+  }
+
+  const backendsList = await apphosting.listBackends(projectId, "-").catch(() => undefined);
+  const backend = backendsList?.backends.find((b) => b.name.split("/").at(-1) === backendId);
```

Comment:
Does `.at(-1)` work in all the supported runtimes, consider using `pop()` instead